lib/src/core/reaction.dart

```diff
@@ -1,27 +1,39 @@
+import 'package:meta/meta.dart';
 import 'package:mobx/src/core/atom.dart';
 import 'package:mobx/src/core/context.dart';
 import 'package:mobx/src/core/derivation.dart';
 
-class TrackingReaction extends Reaction {
-  TrackingReaction(ReactiveContext context, Function() onInvalidate,
+/// Tracks changes that happen between [start] and [end].
+///
+/// This should only be used in situations where it is not possible to
+/// track changes inside a callback function.
+@experimental
+class DerivationTracker {
+  DerivationTracker(ReactiveContext context, Function() onInvalidate,
       {String name})
-      : super(context, onInvalidate, name: name);
+      : _reaction = Reaction(context, onInvalidate, name: name);
```

Comment:
Not clear why we need a `reaction` here for tracking? Could you please elaborate? Is it mostly to wrap the `onInvalidate` in some derivation?